binary_search_tree/binary_search_tree.py

```diff
@@ -1,17 +1,71 @@
 class BinarySearchTree:
-  def __init__(self, value):
-    self.value = value
-    self.left = None
-    self.right = None
+    def __init__(self, value):
+        self.value = value
+        self.left = None
+        self.right = None
 
-  def insert(self, value):
-    pass
+    def insert(self, value):
+        # adds the input value to the binary search tree, adhering to the rules of the ordering of elements in a binary search tree.
+        # Dont forget to wrap the value in a node
+        # 1. compare the element against the current node's value
+        # 2. based on the result of the comparison, go left or right
+        # 3. if we find an empty spot, park the value there
+        # 4. otherwise, go back to step 1
 
```

Comment:
Not sure if you were getting this from the lecture, but this is an almost perfect rendition of how you are supposed to approach this problem. Great work.